components/tidb_query_datatype/src/codec/mysql/json/path_expr.rs

```diff
@@ -556,8 +534,23 @@ mod tests {
                 None,
             ),
             (
-                "$ [ 2147483648 ]",
-                Some("Invalid JSON path expression. The error is around character position 15."),
+                "$ [ 4294967296 ]",
+                Some("Invalid JSON path expression. The error is around character position 5."),
+                None,
+            ),
+            (
+                "$ [ 1to2 ]",
+                Some("Invalid JSON path expression. The error is around character position 2."),
```

Comment:
The position is also a bit strange. Can we lift error to failure after we already match `[`?

Reply:
Yes! I forgot to lift to failure for the ending ']'. It successfully parses an index `1`, but failed to get the `]` (which is actually a `t`).

Now, after lifting, the error position is 6 (the index of `t`). PTAL